From: Armin Novak <armin.novak@thincast.com>
Date: Tue, 30 Nov 2021 08:52:31 +0100
Subject: Fixed missing return check

(cherry picked from commit 46c78cf0f07180258928c16e804aab8bca866a62)
(cherry picked from commit 311558c964626701bd19b2e216cbe85a5df25377)
---
 winpr/libwinpr/sspi/NTLM/ntlm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/winpr/libwinpr/sspi/NTLM/ntlm.c b/winpr/libwinpr/sspi/NTLM/ntlm.c
index 22c8af9..98a130b 100644
--- a/winpr/libwinpr/sspi/NTLM/ntlm.c
+++ b/winpr/libwinpr/sspi/NTLM/ntlm.c
@@ -604,6 +604,9 @@ static SECURITY_STATUS SEC_ENTRY ntlm_InitializeSecurityContextW(
 		{
 			status = ntlm_read_ChallengeMessage(context, input_buffer);
 
+			if (status != SEC_I_CONTINUE_NEEDED)
+				return status;
+
 			if (!pOutput)
 				return SEC_E_INVALID_TOKEN;
 
